navpy/core/navpy.py

```diff
@@ -42,40 +46,51 @@ def angle2dcm(rotAngle1, rotAngle2, rotAngle3, input_unit='rad',
     -----
     Programmer:    Adhika Lie
     Created:    	 May 03, 2011
-    Last Modified: March 06, 2013
+    Last Modified: January 12, 2016
     """
-    R3 = np.zeros((3, 3))
-    R2 = np.zeros((3, 3))
-    R1 = np.zeros((3, 3))
+    rotAngle1, N1 = _input_check_Nx1(rotAngle1)
+    rotAngle2, N2 = _input_check_Nx1(rotAngle2)
+    rotAngle3, N3 = _input_check_Nx1(rotAngle3)
+
+    if(N1 != N2 or N1 != N3):
+        raise ValueError('Inputs are not of same dimensions')
+    if(N1 > 1 and output_type != 'ndarray'):
+        raise ValueError('Matrix output requires scalar inputs')
+
+    R3 = np.zeros((N1, 3, 3))
+    R2 = np.zeros((N1, 3, 3))
+    R1 = np.zeros((N1, 3, 3))
 
     if(input_unit == 'deg'):
         rotAngle1 = np.deg2rad(rotAngle1)
         rotAngle2 = np.deg2rad(rotAngle2)
         rotAngle3 = np.deg2rad(rotAngle3)
 
-    R3[2, 2] = 1.0
-    R3[0, 0] = np.cos(rotAngle1)
-    R3[0, 1] = np.sin(rotAngle1)
-    R3[1, 0] = -np.sin(rotAngle1)
-    R3[1, 1] = np.cos(rotAngle1)
+    R3[:, 2, 2] = 1.0
+    R3[:, 0, 0] = np.cos(rotAngle1)
+    R3[:, 0, 1] = np.sin(rotAngle1)
+    R3[:, 1, 0] = -np.sin(rotAngle1)
+    R3[:, 1, 1] = np.cos(rotAngle1)
 
-    R2[1, 1] = 1.0
-    R2[0, 0] = np.cos(rotAngle2)
-    R2[0, 2] = -np.sin(rotAngle2)
-    R2[2, 0] = np.sin(rotAngle2)
-    R2[2, 2] = np.cos(rotAngle2)
+    R2[:, 1, 1] = 1.0
+    R2[:, 0, 0] = np.cos(rotAngle2)
+    R2[:, 0, 2] = -np.sin(rotAngle2)
+    R2[:, 2, 0] = np.sin(rotAngle2)
+    R2[:, 2, 2] = np.cos(rotAngle2)
 
-    R1[0, 0] = 1.0
-    R1[1, 1] = np.cos(rotAngle3)
-    R1[1, 2] = np.sin(rotAngle3)
-    R1[2, 1] = -np.sin(rotAngle3)
-    R1[2, 2] = np.cos(rotAngle3)
+    R1[:, 0, 0] = 1.0
+    R1[:, 1, 1] = np.cos(rotAngle3)
+    R1[:, 1, 2] = np.sin(rotAngle3)
+    R1[:, 2, 1] = -np.sin(rotAngle3)
+    R1[:, 2, 2] = np.cos(rotAngle3)
 
     if rotation_sequence == 'ZYX':
-        C = R1.dot(R2.dot(R3))
+        C = np.einsum('nij, njk, nkm -> nim', R1, R2, R3)
```

Comment:
Can you add this comment:

    # Note: if R1, R2, and R3 are simply 2D matrices, then this is equivalent to:
    # C = R1.dot(R2.dot(R3))

This will help document why np.einsum is being used only to vectorize this transformation.